decidim-core/lib/decidim/core/engine.rb

```diff
@@ -51,6 +51,19 @@ class Engine < ::Rails::Engine
       isolate_namespace Decidim
       engine_name "decidim"
 
+      initializer "Rails 6 autoloader" do
```

Comment:
Please note the following initializer that actually disables the zeitwerk from any Decidim installation. I have taken the decision of not upgrading the zeitwerk, to keep the pull request smaller , and allow smaller iterations on the upcomming changes.

While running the project using the classic autoloader, the following warning is posted by rails, and is different for each one of the classes.

```
DEPRECATION WARNING: Initialization autoloaded the constants Decidim::LayoutHelper, Decidim::CurrentOrganization, Decidim::StripXForwardedHost, Decidim::Api::QueryType, Decidim::Core::UserType, Decidim::Core::UserGroupType, Decidim::Core::NeedsApiFilterAndOrder, Decidim::Core::ComponentList, Decidim::Core::BaseInputSort, Decidim::Core::HasLocalizedInputSort, Decidim::Core::ComponentInputSort, Decidim::Core::BaseInputFilter, Decidim::Core::HasPublishableInputFilter, Decidim::Core::HasLocalizedInputFilter, Decidim::Core::ComponentInputFilter, Decidim::ParticipatoryProcesses::ParticipatoryProcessType, Decidim::Core::ParticipatorySpaceListBase, Decidim::Core::HasHastaggableInputFilter, Decidim::ParticipatoryProcesses::ParticipatoryProcessInputFilter, Decidim::Core::HasPublishableInputSort, Decidim::ParticipatoryProcesses::ParticipatoryProcessInputSort, Decidim::ParticipatoryProcesses::ParticipatoryProcessList, Decidim::Core::ParticipatorySpaceFinderBase, Decidim::Core::ParticipatorySpaceFinder, Decidim::ParticipatoryProcesses::ParticipatoryProcessFinder, Decidim::Assemblies::AssemblyType, Decidim::Core::ParticipatorySpaceInputFilter, Decidim::Core::ParticipatorySpaceInputSort, Decidim::Core::ParticipatorySpaceList, Decidim::Conferences::ConferenceType, Decidim::Consultations::ConsultationType, Decidim::Initiatives::InitiativeType, Decidim::Core::DecidimType, Decidim::Core::StatisticType, Decidim::Core::OrganizationType, Decidim::Core::UserEntityFinder, Decidim::Core::UserEntityInputSort, Decidim::Core::UserEntityInputFilter, Decidim::Core::UserEntityList, Decidim::ApplicationRecord, UploaderContentTypeValidator, Decidim::ApplicationUploader, Decidim::ImageUploader, Decidim::AvatarUploader, Decidim::UserBaseEntity, Decidim::Api::MutationType, Decidim::EventsManager, ActionText::ContentHelper, ActionText::TagHelper, Decidim::SettingsChange, and Decidim::Comments::CommentCreation.
```

cc @oliverbarnes , @mrcasals

Reply:
Thanks for clarifying, @alecslupu! Come to think of it, this could useful as a comment above the initializer block.

As I mentioned during our troubleshooting session, I don't have an opinion about disabling zeitwerk yet. I do worry a little whether there are any Rails 6-specific changes that rely on it, but so far my googling for caveats or issues people've had in doing it hasn't yielded any hits. 

@mrcasals might have a clearer idea of the impact.

Reply:
I don't have any experience with Rails 6, TBH, so I'm not sure how it might impact. But I'm confident with the test suite, so if we had any problems we should find them thanks to it and thanks to the normal development.

I agree the explanation would work super nice as a comment in the code itself! @alecslupu can you add it, please? 😄